datalad/distributed/drop.py

```diff
@@ -346,6 +346,30 @@ def _drop_dataset(ds, paths, what, reckless, recursive, recursion_limit, jobs):
         return
 
     if paths is not None and paths != [ds.pathobj] and what == 'all':
+        if recursive and reckless == 'kill':
+            # check if any paths contains a subdataset, and if so, drop it to
+            # ensure its not left behind
+            for sub in ds.subdatasets(
+                    # just check for subds at the provided path
+                    path=paths,
+                    state='present',
+                    recursive=recursive,
+                    recursion_limit=recursion_limit,
+                    result_xfm='datasets',
+                    on_failure='ignore',
+                    return_type='generator',
+                    result_renderer='disabled'):
+                if sub is not None:
+                    # there is a subdataset underneath the given path
+                    yield from _drop_dataset(
+                        ds=sub,
+                        # everything underneath the subds can go
+                        paths=None,
+                        what=what,
+                        reckless=reckless,
+                        recursive=False,
```

Comment:
Shouldn't `recursive` and `recursion_limit` be passed on here, too?

Reply:
Seems to me we would run into the same issue one level down otherwise, as the condition for the entire change wouldn't be triggered.

Reply:
Hmm, I think we wouldn't because ``path`` isn't set, in which case it would kill everything it finds under there

Reply:
I have amended the test to show this

Reply:
Thx!